### PR TITLE
SONAR-9448 Sanitize api/qualityprofiles/*

### PR DESCRIPTION
**Pull request by teryk**
- parameters from api/rules/search have **not** been updated –> there are parameters of `api/qualityprofiles/activate_rules` and `api/qualityprofiles/deactivate_rules` that have non camel case parameters. It will be addressed during the sanitization of `api/rules`
- change of behavior of api/qualityprofiles/search will follow in another PR related to [SONAR-8918](https://jira.sonarsource.com/browse/SONAR-8918)

### CODE REVIEW

File: server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/ActivateRuleActionTest.java [hunk 1]
```diff
@@ -107,62 +124,61 @@ public void fail_activate_if_built_in_profile() {
...
-//    assertThat(response.getStatus()).isEqualTo(HttpURLConnection.HTTP_NO_CONTENT);
-//    ArgumentCaptor<RuleActivation> captor = ArgumentCaptor.forClass(RuleActivation.class);
-//    Mockito.verify(ruleActivator).activate(any(DbSession.class), captor.capture(), eq(qualityProfile.getKee()));
-//    assertThat(captor.getValue().getRuleKey()).isEqualTo(ruleKey);
-//    assertThat(captor.getValue().getSeverity()).isEqualTo(Severity.BLOCKER);
-//    assertThat(captor.getValue().getParameters()).containsExactly(entry("key1", "v1"), entry("key2", "v2"));
-//    assertThat(captor.getValue().isReset()).isFalse();
-//  }
+  @Test
+  public void activate_rule_in_default_organization() {
```
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/ActivateRuleActionTest.java]: 👍

File: server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/ActivateRuleActionTest.java [hunk 2]
```diff
@@ -107,62 +124,61 @@ public void fail_activate_if_built_in_profile() {
...
+    ArgumentCaptor<RuleActivation> captor = ArgumentCaptor.forClass(RuleActivation.class);
+    verify(ruleActivator).activate(any(DbSession.class), captor.capture(), any(QProfileDto.class));
+    RuleActivation value = captor.getValue();
+    assertThat(value.getRuleKey()).isEqualTo(ruleKey);
+    assertThat(value.getSeverity()).isEqualTo(Severity.BLOCKER);
+    assertThat(value.isReset()).isFalse();
+  }
+
+  @Test
+  public void activate_rule_in_specific_organization() {
```
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/ActivateRuleActionTest.java]: 👍

File: server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/QProfilesWsMediumTest.java [hunk 3]
```diff
@@ -52,14 +52,17 @@
 import org.sonar.server.tester.ServerTester;
```
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/QProfilesWsMediumTest.java]: I will be very joyful when we'll get rid of this ServerTester usage

File: server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/ChangelogActionMockTest.java [hunk 4]
```diff
@@ -45,35 +44,35 @@
...
 import static org.sonar.api.utils.DateUtils.parseDate;
 import static org.sonar.server.qualityprofile.QProfileTesting.XOO_P1_KEY;
-import static org.sonar.server.qualityprofile.ws.ChangelogAction.PARAM_SINCE;
-import static org.sonar.server.qualityprofile.ws.ChangelogAction.PARAM_TO;
-import static org.sonarqube.ws.client.qualityprofile.QualityProfileWsParameters.PARAM_PROFILE_KEY;
+import static org.sonarqube.ws.client.qualityprofile.QualityProfileWsParameters.PARAM_PROFILE;
+import static org.sonarqube.ws.client.qualityprofile.QualityProfileWsParameters.PARAM_SINCE;
+import static org.sonarqube.ws.client.qualityprofile.QualityProfileWsParameters.PARAM_TO;
 
 public class ChangelogActionMockTest {
```
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/ChangelogActionMockTest.java]: Could we rename this class ChangelogActionTest ? Do we need the "Mock" in the class name ?

Comment (teryk) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/ChangelogActionMockTest.java]: we could but there's a competing `ChangelogDatabaseActionTest`… I know, very ugly. I have let is as is for the time being

**Pull request merged**